mmcv/cnn/rfsearch/search.py

```diff
@@ -0,0 +1,263 @@
+        assert mode in ['search', 'fixed_single_branch', 'fixed_multi_branch']
+        assert config is not None
+        self.config = config
+        self.config['structure'] = {}
+        self.verbose = verbose
+        if rfstructure_file is not None:
+            rfstructure = mmcv.load(rfstructure_file)['structure']
+            self.config['structure'] = rfstructure
+        self.mode = mode
+        self.num_branches = self.config['search']['num_branches']
+        self.by_epoch = by_epoch
+
+    def init_model(self, model: nn.Module):
+        """init model with search ability.
+
+        Args:
+            model (nn.Module): pytorch model
+
+        Raises:
+            NotImplementedError: only support three modes:
+                search/fixed_single_branch/fixed_multi_branch
+        """
+        if self.verbose:
+            logger.info('RFSearch init begin.')
+        if self.mode == 'search':
+            if self.config['structure']:
+                self.set_model(model, search_op='Conv2d')
+            self.wrap_model(model, search_op='Conv2d')
+        elif self.mode == 'fixed_single_branch':
+            self.set_model(model, search_op='Conv2d')
+        elif self.mode == 'fixed_multi_branch':
+            self.set_model(model, search_op='Conv2d')
+            self.wrap_model(model, search_op='Conv2d')
+        else:
+            raise NotImplementedError
+        if self.verbose:
+            logger.info('RFSearch init end.')
+
+    def after_train_epoch(self, runner):
+        """Do search after one training epoch.
+
+        Args:
+            runner (_type_): MMCV runner
+        """
+        if self.by_epoch and self.mode == 'search':
+            self.step(runner.model, runner.work_dir)
+
+    def after_train_iter(self, runner):
+        """Do search after one training iteration.
+
+        Args:
+            runner (_type_): MMCV runner
+        """
+        if not self.by_epoch and self.mode == 'search':
+            self.step(runner.model, runner.work_dir)
+
+    def step(self, model: nn.Module, work_dir: str):
+        """Performs a dilation searching step.
+
+        Args:
+            model (nn.Module): pytorch model
+            work_dir (str): save path
+        """
+        self.config['search']['step'] += 1
+        if (self.config['search']['step']
+            ) % self.config['search']['search_interval'] == 0 and (self.config[
+                'search']['step']) < self.config['search']['max_step']:
+            self.search(model)
+            for name, module in model.named_modules():
+                if isinstance(module, ConvRFSearchOp):
+                    self.config['structure'][name] = module.op_layer.dilation
+            write_to_json(
+                self.config,
+                os.path.join(
+                    work_dir,
+                    'local_search_config_step%d.json' %
+                    self.config['search']['step'],
+                ),
+            )
```

Comment:
```suggestion
            for module in model.modules():
                if isinstance(module, ConvRFSearchOp):
                    self.config['structure'][name] = module.op_layer.dilation

            write_to_json(
                self.config,
                os.path.join(
                    work_dir,
                    'local_search_config_step%d.json' %
                    self.config['search']['step'],
                ),
            )
```

Reply:
The 'name' is used in line 137, so ‘named_modules’ should be used.

Reply:
Got it.